Library/Homebrew/dev-cmd/bottle.rb

```diff
@@ -359,18 +378,24 @@ def gnu_tar_formula_ensure_installed_if_needed!
         gnu_tar_formula
       end
 
-      sig { params(mtime: String, default_tar: T::Boolean).returns([String, T::Array[String]]) }
-      def setup_tar_and_args!(mtime, default_tar: false)
+      sig { params(mtime: String).returns([String, T::Array[String]]) }
+      def setup_tar_and_args!(mtime)
         # Without --only-json-tab bottles are never reproducible
         default_tar_args = ["tar", tar_args].freeze
-        return default_tar_args if !args.only_json_tab? || default_tar
+        return default_tar_args unless args.only_json_tab?
 
         # Use gnu-tar as it can be set up for reproducibility better than libarchive
         # and to be consistent between macOS and Linux.
-        gnu_tar_formula = gnu_tar_formula_ensure_installed_if_needed!
-        return default_tar_args if gnu_tar_formula.blank?
+        tar = if system_gnu_tar?
+          "tar"
```

Comment:
Simplest way to update this PR is probably following with additional input parameter for `formula`:
```suggestion
        tar = if formula.name == "gnu-tar" && system_gnu_tar?
          "tar"
```

Though this won't handle case if a user runs `brew bottle gnu-tar` on an old LTS Linux. Seems like a very unlikely situation.

---

A more complete option would be to add OS-specific logic to only check this on Linux-only as macOS doesn't have the same problem in relation to rewritten `ld.so`.

---

Either way, if we do want to consider pure Ruby alternative, then these changes would be a stopgap measure.

Reply:
This seems fine to me 👍🏻